lib/tbot/config/output_database.go

```diff
@@ -0,0 +1,150 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gravitational/trace"
+	"golang.org/x/exp/slices"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/tbot/bot"
+	"github.com/gravitational/teleport/lib/tbot/identity"
+)
+
+const DatabaseOutputType = "database"
+
+type DatabaseSubtype string
+
+var (
+	// UnspecifiedDatabaseSubtype is the unset value and the default. This
+	// should work for most databases.
+	UnspecifiedDatabaseSubtype DatabaseSubtype = ""
+	// TLSDatabaseSubtype is for databases that require specifically named
+	// outputs: tls.key, tls.crt and tls.cas
+	TLSDatabaseSubtype DatabaseSubtype = "tls"
+	// MongoDatabaseSubtype indicates credentials should be generated which
+	// are compatible with MongoDB.
+	MongoDatabaseSubtype DatabaseSubtype = "mongo"
+	// CockroachDatabaseSubtype indicates credentials should be generated which
+	// are compatible with CockroachDB.
+	CockroachDatabaseSubtype DatabaseSubtype = "cockroach"
+
+	databaseSubtypes = []DatabaseSubtype{
+		UnspecifiedDatabaseSubtype,
+		TLSDatabaseSubtype,
+		MongoDatabaseSubtype,
+		CockroachDatabaseSubtype,
+	}
+)
+
+// DatabaseOutput produces credentials which can be used to connect to a
+// database through teleport.
+type DatabaseOutput struct {
+	Common OutputCommon `yaml:",inline"`
+	// Subtype indicates the type of the database you are generating credentials
+	// for. An empty value is supported by most database, but CockroachDB and
+	// MongoDB require this value to be set to `mongo` and `cockroach`
+	// respectively.
+	Subtype DatabaseSubtype `yaml:"subtype,omitempty"`
+	// Service is the service name of the Teleport database. Generally this is
+	// the name of the Teleport resource. This field is required for all types
+	// of database.
+	Service string `yaml:"service"`
+	// Database is the name of the database to request access to.
+	Database string `yaml:"database,omitempty"`
+	// Username is the database username to request access as.
+	Username string `yaml:"username,omitempty"`
+}
+
+func (o *DatabaseOutput) templates() []template {
+	templates := []template{
+		&templateTLSCAs{},
+		&templateIdentity{},
+	}
+	if o.Subtype == MongoDatabaseSubtype {
+		templates = append(templates, &templateMongo{})
+	}
+	if o.Subtype == CockroachDatabaseSubtype {
+		templates = append(templates, &templateCockroach{})
+	}
+	if o.Subtype == TLSDatabaseSubtype {
+		templates = append(templates, &templateTLS{
+			caCertType: types.HostCA,
```

Comment:
I'm not sure that the HostCA is the correct CA to be outputting here, but this is the behaviour that users would get if they currently followed our documentation and wrote something like:
```
destinations:
  - directory: /opt/machine-id
    
    database:
      service: example-server
      username: alice
      database: example
    
    configs:
      - tls
```

 I'm split on if we should "fix" this here, or maintain the behaviour.

Reply:
Further testing reveals this is the correct CA to be emitting.